src/modules/dwm.cpp

```diff
@@ -0,0 +1,493 @@
+        return true;
+      }
+      m_log.info("%s: Building floating label", name());
+      builder->node(m_floating_label);
+    } else if (tag == TAG_LABEL_LAYOUT) {
+      m_log.info("%s: Building layout label", name());
+      if (m_layout_click) {
+        // Toggle between secondary and default layout
+        auto addr = (m_current_layout == m_default_layout ? m_secondary_layout : m_default_layout)->address;
+        builder->cmd(mousebtn::LEFT, build_cmd(CMD_LAYOUT_SET, to_string(addr)));
+        // Set previous layout
+        builder->cmd(mousebtn::RIGHT, build_cmd(CMD_LAYOUT_SET, "0"));
+      }
+      if (m_layout_scroll) {
+        auto addr_next = next_layout(*m_current_layout, m_layout_wrap)->address;
+        auto addr_prev = prev_layout(*m_current_layout, m_layout_wrap)->address;
+        // Set address based on scroll direction
+        auto scroll_down_addr = to_string(m_layout_reverse ? addr_prev : addr_next);
+        auto scroll_up_addr = to_string(m_layout_reverse ? addr_next : addr_prev);
+        builder->cmd(mousebtn::SCROLL_DOWN, build_cmd(CMD_LAYOUT_SET, scroll_down_addr));
+        builder->cmd(mousebtn::SCROLL_UP, build_cmd(CMD_LAYOUT_SET, scroll_up_addr));
+      }
+      builder->node(m_layout_label);
+      if (m_layout_click) {
+        builder->cmd_close();
+        builder->cmd_close();
+      }
+      if (m_layout_scroll) {
+        builder->cmd_close();
+        builder->cmd_close();
+      }
+    } else if (tag == TAG_LABEL_TAGS) {
+      m_log.info("%s: Building tags label", name());
+      bool first = true;
+      for (const auto& tag : m_tags) {
+        // Don't insert separator before first tag
+        if (first) {
+          first = false;
+        } else if (*m_seperator_label) {
+          builder->node(m_seperator_label);
+        }
+
+        if (m_tags_click) {
+          builder->cmd(mousebtn::LEFT, build_cmd(CMD_TAG_VIEW, to_string(tag.bit_mask)));
+          builder->cmd(mousebtn::RIGHT, build_cmd(CMD_TAG_TOGGLE_VIEW, to_string(tag.bit_mask)));
+          builder->node(tag.label);
+          builder->cmd_close();
+          builder->cmd_close();
+        } else {
+          builder->node(tag.label);
+        }
+      }
+    } else {
+      return false;
+    }
+    return true;
+  }
+
+  auto dwm_module::check_send_cmd(string cmd, const string& ipc_cmd) -> bool {
+    // cmd = <EVENT_PREFIX><ipc_cmd>-<arg>
+    cmd.erase(0, strlen(EVENT_PREFIX));
+
+    // cmd = <ipc_cmd>-<arg>
+    if (cmd.compare(0, ipc_cmd.size(), ipc_cmd) == 0) {
+      // Erase '<ipc_cmd>-'
+      cmd.erase(0, ipc_cmd.size() + 1);
+      m_log.info("%s: Sending workspace %s command to ipc handler", name(), ipc_cmd);
+
+      try {
+        m_ipc->run_command(ipc_cmd, (Json::UInt64)stoul(cmd));
+        return true;
+      } catch (const dwmipc::SocketClosedError& err) {
+        m_log.err("%s: Disconnected from socket: %s", name(), err.what());
+        sleep(chrono::duration<double>(1));
```

Comment:
Same here. This always blocks the entire bar because `input` is called by the controller

Reply:
I'll look into it

Reply:
See commits 4a5070ce6c8a152be0945afb7d6bde2ed269315f and b1caed225b660db2edc420e03e504af5650d660c